packages/core/src/api/blockManipulation/commands/replaceBlocks/replaceBlocks.ts

```diff
@@ -1,15 +1,165 @@
-import type { Node } from "prosemirror-model";
-import type { Transaction } from "prosemirror-state";
+import { ResolvedPos, Slice, type Node } from "prosemirror-model";
+import { TextSelection, type Transaction } from "prosemirror-state";
+import { ReplaceAroundStep } from "prosemirror-transform";
 import type { Block, PartialBlock } from "../../../../blocks/defaultBlocks.js";
 import type {
   BlockIdentifier,
   BlockSchema,
   InlineContentSchema,
   StyleSchema,
 } from "../../../../schema/index.js";
+import { getBlockInfoFromResolvedPos } from "../../../getBlockInfoFromPos.js";
 import { blockToNode } from "../../../nodeConversions/blockToNode.js";
 import { nodeToBlock } from "../../../nodeConversions/nodeToBlock.js";
 import { getPmSchema } from "../../../pmUtil.js";
+import {
+  getParentBlockInfo,
+  getPrevBlockInfo,
+} from "../mergeBlocks/mergeBlocks.js";
+
+// TODO: Where should this function go?
+/**
+ * Moves the first block in a column to the previous/next column and handles
+ * all necessary collapsing of `column`/`columnList` nodes. Only moves the
+ * block to the start of the next column if it's in the first column.
+ * Otherwise, moves the block to the end of the previous column.
+ * @param tr The transaction to apply changes to.
+ * @param blockBeforePos The position just before the first block in the column.
+ * @returns The position just before the block, after it's moved.
+ */
+export function moveFirstBlockInColumn(
+  tr: Transaction,
+  blockBeforePos: ResolvedPos,
+): ResolvedPos {
+  const blockInfo = getBlockInfoFromResolvedPos(blockBeforePos);
+  if (!blockInfo.isBlockContainer) {
+    throw new Error(
+      "Invalid blockBeforePos passed to moveFirstBlockInColumn: does not point to blockContainer node.",
+    );
```

Comment:
```suggestion
    throw new Error(
      "Invalid blockBeforePos: does not point to blockContainer node.",
      { cause: { blockBeforePos } }
    );
```

We should standardize errors & error messages in the future, but I'm liking this sort of pattern